st2client/st2client/shell.py

```diff
@@ -241,6 +242,10 @@ def get_client(self, args, debug=False):
             try:
                 token = self._get_auth_token(client=client, username=username, password=password,
                                              cache_token=cache_token)
+            except requests.exceptions.ConnectionError as e:
+                LOG.warn('API server is not available, skipping authentication.')
+                LOG.exception(e)
+                return client
```

Comment:
I do see it. As a client, we have no idea if server is auth enabled or not. It looks like there needs to a handshake at the start to discover server capabilities. I am not proposing we do it now but it looks like we might soon need it.